bsb/connectivity/point_cloud/geometric_shapes.py

```diff
@@ -0,0 +1,1205 @@
+        :rtype: Tuple(numpy.ndarray[float], numpy.ndarray[float], numpy.ndarray[float]) | None
+        """
+        if len(self._shapes) != 0:
+            cloud = np.empty([3])
+            npoints = self.compute_n_points()
+            x = []
+            y = []
+            z = []
+            for shape in self._shapes:
+                xt, yt, zt = shape.wireframe_points()
+                x.append(xt)
+                y.append(yt)
+                z.append(zt)
+            return x, y, z
+        else:
+            return None
+
+    def inside_mbox(self, points: numpy.ndarray[float]) -> numpy.ndarray[bool]:
+        """
+        Check if the points given in input are inside the minimal bounding box of the collection.
+
+        :param numpy.ndarray points: An array of 3D points.
+        :return: A bool np.ndarray specifying whether each point of the input array is inside the
+            minimal bounding box of the collection.
+        :rtype: numpy.ndarray[bool]
+        """
+        inside = (
+            (points[:, 0] > self._mbb_min[0])
+            & (points[:, 0] < self._mbb_max[0])
+            & (points[:, 1] > self._mbb_min[1])
+            & (points[:, 1] < self._mbb_max[1])
+            & (points[:, 2] > self._mbb_min[2])
+            & (points[:, 2] < self._mbb_max[2])
+        )
+
+        return inside
+
+    def inside_shapes(self, points: numpy.ndarray[float]) -> numpy.ndarray[bool]:
+        """
+        Check if the points given in input are inside at least in one of the shapes of the
+        collection.
+
+        :param numpy.ndarray points: An array of 3D points.
+        :return: A bool numpy.ndarray specifying whether each point of the input array is inside the
+            collection of shapes or not.
+        :rtype: numpy.ndarray[bool]
+        """
+        if len(self._shapes) != 0:
+            cloud = np.full(len(points), 0, dtype=bool)
+            for shape in self._shapes:
+                tmp = shape.check_mbox(points)
+                if np.any(tmp):
+                    cloud = cloud | shape.check_inside(points)
+            return cloud
+        else:
+            return None
+
+    def save_to_file(self, filename: str):
+        """
+        Save the collection of shapes to file.
+
+        :param str filename: The name of the output file.
+        """
+        with open(filename, "wb") as handle:
+            pickle.dump(self, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+    def load_from_file(self, filename: str):
+        """
+        Load a collection of shapes from a file.
+
+        :param str filename: The name of the output file.
+        """
+        tmp = None
+        with open(filename, "rb") as handle:
+            tmp = pickle.load(handle)
+        self._shapes = tmp._shapes
+        self._voxel_size = tmp._voxel_size
+        self._mbb_min, self._mbb_max = self.find_mbb()
+        self._labels = tmp._labels
+
+
+class Ellipsoid(GeometricShape):
```

Comment:
```suggestion
@config.node
class Ellipsoid(GeometricShape, classmap_entry="ellipsoid"):
        center = config.attr(type=types.ndarray(dtype=float), required=True)
        ...
```

Class map entry may be omitted and it will snake case by default, but I think writing it is clearer and done everywhere else in the code.